kernel/init.c

```diff
@@ -253,6 +253,13 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 
 	main();
 
+#ifdef CONFIG_COVERAGE_HW_TEST
```

Comment:
put the ifdef somewhere else where the implementation is done and create noop functions for when the config is not eanbled